apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -18,13 +18,123 @@
 from werkzeug.wrappers import Request
 import common, errno, getopt, json, logging, os, socket, string, sys, time
 
-numeric_project_id = 12345
-numeric_project_id_token = u'<NUMERIC_PROJECT_ID>'
-project_id = u'fake-project'
-project_id_token = u'<PROJECT_ID>'
+class ServerState:
+  _credential_scopes = ["https://www.googleapis.com/auth/cloud-platform",
+                        "https://www.googleapis.com/auth/userinfo.email"]
 
-credential_scopes = [ "https://www.googleapis.com/auth/cloud-platform", "https://www.googleapis.com/auth/userinfo.email" ]
-credentials = {}
+  def __init__(self, project_id, numeric_project_id, use_default_credentials, credentials_files):
+    self.project_id = project_id
+    self.numeric_project_id = numeric_project_id
+    
+    self._credentials = {}
+    if use_default_credentials:
+      self._useDefaultCredentials()
+    if credentials_files:
+      self._useCredentials(credentials_files)
+
+    self._loadData()
+
+  def _useCredentials(self, credentials_files):
+    for path in string.split(credentials_files, os.pathsep):
+      account_credentials = service_account.ServiceAccountCredentials.from_json_keyfile_name(
+        path,
+        scopes = ServerState._credential_scopes)
+      self._credentials[account_credentials.service_account_email] = account_credentials
+
+  def _useDefaultCredentials(self):
+    self._credentials[u'default'] = service_account.ServiceAccountCredentials.get_application_default()
+
+  def _loadData(self):
```

Comment:
Once the design solidifies a bit, I really hope we can get some comments in here... I think it'd be really helpful. Not going to block on that because right now it's just the couple of you and you're still evolving it, but I think someone else would struggle to figure out what is going on without comments